### PR TITLE
Bump nixpkgs, bump GHC to 9.6.6 and stackage to LTS 22.34

### PR DESCRIPTION
**Pull request by tbagrel1**
This pull request changes the default version of GHC to be the one used by the latest stackage LTS available at the moment.

### CODE REVIEW

File: linear-base.cabal [hunk 1]
```diff
@@ -21,13 +21,26 @@ source-repository head
...
-common warnings
-    ghc-options: -Wall -Wcompat -Wincomplete-record-updates -Wincomplete-uni-patterns -Wnoncanonical-monad-instances
+common build-opts
+    ghc-options: -O2 -Wall -Wcompat -Wincomplete-record-updates -Wincomplete-uni-patterns -Wnoncanonical-monad-instances
                  -- Additional warnings we may consider adding:
                  -- * -Wredundant-constraints : would need deactivating in the modules which use Nat
+common rts-opts-multithread
+    ghc-options: -threaded -rtsopts -with-rtsopts=-N
+common rts-opts-monothread
+    ghc-options: -threaded -rtsopts -with-rtsopts=-N1
```
Comment (aspiwack) [inline linear-base.cabal:31]: Why not simply removing `-threaded`?
```suggestion
    ghc-options: ""
```

File: src/Data/Unrestricted/Linear/Internal/Instances.hs [hunk 2]
```diff
@@ -186,7 +186,8 @@ instance (KnownNat n, Consumable a) => Consumable (V n a) where
 
 instance (KnownNat n, Dupable a) => Dupable (V n a) where
   dupR (V xs) =
-    V . Unsafe.toLinear (Vector.fromListN (V.theLength @n))
+    V
+      . Unsafe.toLinear (Vector.fromListN (V.theLength @n))
```
Comment (aspiwack) [inline src/Data/Unrestricted/Linear/Internal/Instances.hs:190]: What happened here?

File: linear-base.cabal (hunk 1 above)
Comment (tbagrel1) [reply linear-base.cabal:31]: Apparently there are [some non-negligible](https://discourse.haskell.org/t/make-ghc-threaded-the-default-rts/4220/6) differences, which indicates we should probably use no option at all for the bench case.

File: src/Data/Unrestricted/Linear/Internal/Instances.hs (hunk 2 above)
Comment (tbagrel1) [reply src/Data/Unrestricted/Linear/Internal/Instances.hs:190]: Our ormolu version is given by stackage AFAIK, which I bumped.
I suppose the new ormolu version has some breaking changes related to default operator fixity/formatting.
(I didn't change that manually)

**Review by aspiwack — APPROVED** [review]

**Pull request merged**